jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/editor/S3TreeTable.kt

```diff
@@ -119,16 +126,19 @@ class S3TreeTable(
                 }
             }
         }
+        return true
     }
 
-    private fun handleLoadingMore(row: Int) {
-        val continuationNode = (tree.getPathForRow(row).lastPathComponent as? DefaultMutableTreeNode)?.userObject as? S3TreeContinuationNode ?: return
-        val parent = continuationNode.parent ?: return
+    private fun handleLoadingMore(row: Int): Boolean {
+        val continuationNode = (tree.getPathForRow(row).lastPathComponent as? DefaultMutableTreeNode)?.userObject as? S3TreeContinuationNode ?: return false
+        val parent = continuationNode.parent ?: return false
 
-        ApplicationManager.getApplication().executeOnPooledThread {
+        GlobalScope.launch {
```

Comment:
?

Reply:
The rest of the class uses global scope, this will need to be refactored eventually